wasm/execution/configuration.py

```diff
@@ -125,6 +129,23 @@ def __init__(self, store: Store) -> None:
         super().__init__(store)
         self.frame_stack = FrameStack()
 
+    def execute(self) -> Tuple[TValue, ...]:
+        # TODO: unwrap import once logic functions move out of wasm.main
+        from wasm.main import opcode2exec
+
+        while self.has_active_frame:
+            instruction = next(self.instructions)
+
+            logic_fn = opcode2exec[instruction.opcode][0]
```

Comment:
I can't remember why there's a `[0]` at the end here, or what is in the later indices. Sorry, no suggestions, just identifying a part that wasn't immediately apparent.

Reply:
This is the old/legacy implementation still bleading through.  It's setup as `Dict[BinaryOpcode, Tuple[Callable, ...]]`.  Most of the opcodes are a single function (thus the `[0]` indexing).  The numeric operations are done as two callables, one that handles popping values off the stack and then passing the values into the second function.

This pattern is going away with each opcode just mapping to one logic function.